addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/internal/OneWireGPIOSensorReader.java

```diff
@@ -0,0 +1,43 @@
+package org.openhab.binding.onewiregpio.internal;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OneWireGPIOSensorReader {
+
+    private static Logger logger = LoggerFactory.getLogger(OneWireGPIOSensorReader.class);
+
+    private static String FILE_TEMP_MARKER = "t=";
+
+    private static BigDecimal INVALID_TEMPERATURE = BigDecimal.valueOf(-127);
+
+    public static BigDecimal readSensorTemperature(String gpioFile) {
```

Comment:
this is all static - it might make sense to simply move this method into the handler and remove the class.